### PR TITLE
Unify TextChange chars into strings

### PR DESCRIPTION
**Pull request by oferb**

### CODE REVIEW

File: common/TextDifferencer.java [hunk 1]
```diff
@@ -51,7 +52,34 @@
...
+    TextChange previous = null;
+    for (TextChange current : changes) {
+      System.out.println(current);
+      if (previous == null) {
+        previous = current;
+        unified = current;
+        continue;
+      }
+      if (previous.getType() == current.getType()) {
+        unified = unified.toBuilder().setDifference(unified.getDifference() + current.getDifference()).build();
```
Comment (yanivsn) [inline common/TextDifferencer.java]: If you will work with builders directly, you can avoid changing back and forth from builder to protobuf.

File: common/TextDifferencer.java [hunk 2]
```diff
@@ -34,7 +35,7 @@
    * @param secondString The second string.
    * @return A list of text changes.
    */
-  public static List<TextChange> getAllTextChanges(String firstString, String secondString) {
+  public static ImmutableList<TextChange> getAllTextChanges(String firstString, String secondString) {
```
[the remark this answers is not in the document]
Comment (jbduncan) [reply common/TextDifferencer.java]: @yanivsn True, but the Guava team recommend treating `ImmutableList` as if it were a specialised interface, so unless there's a good reason not to return `ImmutableList` (e.g. the returned list is expected to be mutable, or you expect to use an alternative immutable collections library like Vavr), then I'd recommend keeping this signature change from `List` to `ImmutableList`. :)

[the remark this answers is not in the document]
Comment (jbduncan) [reply common/TextDifferencer.java]: More info can be found at http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/ImmutableCollection.html :)

File: common/TextDifferencer.java (hunk 1 above)
Comment (oferb) [reply common/TextDifferencer.java]: Done

File: common/TextDifferencer.java [hunk 3]
```diff
@@ -51,7 +52,34 @@
     allChanges.addAll(
         getMatchingTextChanges(
             first, first.length - footerLength, second.length - footerLength, footerLength));
-    return allChanges;
+    return unifyTextChanges(ImmutableList.copyOf(allChanges));
```
[the remark this answers is not in the document]
Comment (oferb) [reply common/TextDifferencer.java]: As discussed, we'll pass a list. `unifyTextChanges()` is private so our API is still using `ImmutableList`.

File: common/TextDifferencer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (oferb) [reply common/TextDifferencer.java]: Thanks @jbduncan!

**Pull request merged**